SwrveCommonSDK/src/main/java/com/swrve/sdk/SwrveBase.java

```diff
@@ -235,6 +236,9 @@ public void onMessage(final SwrveConversation conversation) {
                                 intent.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
                                 intent.putExtra("conversation", conversation);
                                 ctx.startActivity(intent);
+
+                                // Report that the message was shown to users
+                                conversation.getCampaign().messageWasShownToUser();
```

Comment:
That method does not send an impression event. It increments impressions internally (a counter used to check later on) and sets the min message delay